### PR TITLE
router: Fix race registering for service discovery

### LINKED ISSUE #917: Bootstrap gitreceive-route error
```text
bootstrap ===> gitreceive-route start
bootstrap ===> gitreceive-route error
unknown_error: Something went wrong
```

And in the controller log:

```text
t=2015-01-31T23:44:28+0000 lvl=info msg="request started" component=controller req_id=b7d1f64ffa9144fa994b61c6bb09bafe method=POST path=/apps/4e5464d2e9e74feb9fb74dbc9291abf9/routes client_ip=100.100.69.1
t=2015-01-31T23:44:28+0000 lvl=eror msg="Post http://router-api.discoverd:5000/routes: dial tcp 10.69.84.5:5000: connection refused" component=controller req_id=b7d1f64ffa9144fa994b61c6bb09bafe
t=2015-01-31T23:44:28+0000 lvl=info msg="request completed" component=controller req_id=b7d1f64ffa9144fa994b61c6bb09bafe status=500 duration=2.940227ms
```

https://s3.amazonaws.com/flynn-ci-logs/20150131234209-af6a5fa4-build-1e6e6e550e48616b0d05e1bc496b76faa902a73f-2015-01-31-23-44-29.txt

### PR DESCRIPTION
**Pull request by titanous**
Don’t register with service discovery until all of the listeners have successfully started, otherwise bootstrap may race to connect.

Closes #917

### CODE REVIEW

File: router/server.go [hunk 1]
```diff
@@ -140,5 +128,18 @@ func main() {
 	if err != nil {
 		shutdown.Fatal(err)
 	}
+
```
Comment (heavenlyhash) [inline router/server.go]: There's another `ListenAndServe` a few lines above the fold in a goroutine, should that be split to ensure bind before discoverd annouce too?

Comment (titanous) [inline router/server.go]: Yeah, will do.

Comment (titanous) [inline router/server.go]: Done.

**Comment by lmars** [comment]
LGTM

**Pull request merged**